test/integration/arbitrum-mainnet/Loans.fork.t.sol

```diff
@@ -141,48 +161,85 @@ contract LoansForkTest is LoansTestBase {
         (uint newLoanId, uint newLoanAmount, int transferAmount) =
             rollLoan(pair, user, loanId, rollOfferId, minToUser);
 
+        // Verify fee taken and sent to recipient
+        assertEq(
+            pair.cashAsset.balanceOf(feeRecipient) - feeRecipientBalanceBefore, expectedResults.protocolFee
+        );
+        assertEq(int(pair.cashAsset.balanceOf(provider) - providerBalanceBefore), expectedResults.toProvider);
         assertGt(newLoanId, loanId);
         assertGe(int(newLoanAmount), int(initialLoanAmount) + transferAmount);
     }
 
     function testFullLoanLifecycle() public {
+        uint feeRecipientBalanceBefore = pair.cashAsset.balanceOf(feeRecipient);
+
         uint offerId = createProviderOffer(pair, callstrikeToUse, offerAmount);
-        uint minLoanAmount = 0.3e6;
+        uint initialFee = getProviderProtocolFeeByUnderlying();
+
         (uint loanId, uint providerId, uint initialLoanAmount) =
-            openLoan(pair, user, underlyingAmount, minLoanAmount, offerId);
+            openLoan(pair, user, underlyingAmount, 0.3e6, offerId);
 
-        // Advance time to simulate passage of time
+        // Verify fee taken and sent to recipient
+        assertEq(pair.cashAsset.balanceOf(feeRecipient) - feeRecipientBalanceBefore, initialFee);
+
+        // Advance time
         vm.warp(block.timestamp + pair.durations[0] - 20);
 
+        uint recipientBalanceAfterOpenLoan = pair.cashAsset.balanceOf(feeRecipient);
         uint rollOfferId = createRollOffer(pair, provider, loanId, providerId, rollFee, rollDeltaFactor);
 
-        int minToUser = -1000e6;
+        // Calculate roll protocol fee
+        IRolls.PreviewResults memory expectedResults =
+            pair.rollsContract.previewRoll(rollOfferId, pair.takerNFT.currentOraclePrice());
+
         (uint newLoanId, uint newLoanAmount, int transferAmount) =
-            rollLoan(pair, user, loanId, rollOfferId, minToUser);
+            rollLoan(pair, user, loanId, rollOfferId, -1000e6);
+
+        assertEq(
+            pair.cashAsset.balanceOf(feeRecipient) - recipientBalanceAfterOpenLoan,
+            expectedResults.protocolFee
+        );
 
         // Advance time again
         vm.warp(block.timestamp + pair.durations[0]);
 
-        uint minUnderlyingOut = underlyingAmount * 95 / 100; // 5% slippage
+        uint minUnderlyingOut = underlyingAmount * (100 - slippage) / 100; // 5% slippage
         uint underlyingOut = closeLoan(pair, user, newLoanId, minUnderlyingOut);
 
+        // Verify total protocol fees collected
+        assertEq(
+            pair.cashAsset.balanceOf(feeRecipient) - feeRecipientBalanceBefore,
+            initialFee + expectedResults.protocolFee
+        );
+
         assertGt(initialLoanAmount, 0);
-        assertGt(newLoanId, loanId);
         assertGe(int(newLoanAmount), int(initialLoanAmount) + transferAmount);
         assertGe(underlyingOut, minUnderlyingOut);
```

Comment:
this checks the return value and not the balance, so a bit less robust

Reply:
changed it